### PR TITLE
Placeholder for the FQDN of Remote R2W

### PR DESCRIPTION
**Pull request by stephdl**
- When the property openvpn@host-to-net{'Remote'} is empty (first configuration) we display all external IP
- If you erase all of these IP, the openvpn@host-to-net{'Remote'} is saved empty
- If you download the R2W VPN configuration you will find that the used remote is the FQDN of the server
- If you go back to the configuration of the R2W VPN you will see the external IPs, but the configuration of the R2W sent to your clients use the FQDN
 

I propose to change a bit this behavior

openvpn@host-to-net{'Remote'} is empty  (first configuration) -> display all external IP in the textarea
the textarea of openvpn@host-to-net{'Remote'} is empty -> display a placeholder with the FQDN of the server
the textarea of openvpn@host-to-net{'Remote'} is saved empty -> save to openvpn@host-to-net{'Remote'} the FQDN of the server

https://github.com/NethServer/dev/issues/6337

### CODE REVIEW

**Review by cotosso — APPROVED** [review]
From the user's point of view it makes it clearer which value is used, it works as expected and it's more intuitive.

**Review by gsanchietti — CHANGES_REQUESTED** [review]
While the place holder looks good, the FQDN saved inside the db is not.

If the admin changes the FQDN of the machine, than creates a new client and send the config to the user, the OpenVPN will fail the connection because of the wrong hostname.
The admin is not aware of such change until it opens the `Roadwarrior server` configuration.

**Comment by stephdl** [comment]
so lets close it

what do you think @cotosso ?

**Comment by gsanchietti** [comment]
I think you should search for an alternative solution: do not store the FQDN, but add a button in the UI like "Retrieve public IP".
When the button is clicked, the public IP is copied inside the "Remote" field.

If the remote field is empty, do nothing, just display the current placeholder with the FQDN.

**Comment by stephdl** [comment]
I wonder if we are not adding complex things to a complex thing.

If you save an IP to the remote property, you can also change the public IP one day.
If you save a FDQN to the remote property, you might change it also one day.

All the two possibilities will bring probably some noises but changing the external IP or the FQDN of a server is always a bad decision. At this point I would like to do nothing more if really you think about bad changes.

**Comment by cotosso** [comment]
I analyzed @stephdl solution and @gsanchietti  observations, I would take a step back by focusing on the reported problem.

Currently the panel has a bug that we must try to fix

* Editing server roadwarrior,  in the box “Contact this server on public IP / host", the administrator can delete the pre-compiled IPs and save the configuration without any value of remote(in this way it will uses the FQDN as remote)

* Returning to the configuration page, cause the value is empty, the box shows again the IP addresses that the admin had already previously deleted and this is highly misleading. 
It can also push the sysadmin to make mistakes (just click on "save" without deleting the IPs and the OpenVPN client export will have IP addresses instead of FQDN). 
Consider that as soon as you enter the vpn server in editing mode you will see the IP addresses and **you no longer will know if those are the values ​​you wrote or if they were put in the box by the GUI because the field is empty.** 


@stephdl solution makes configuration clearer to the admin, but loose dynamic modification of the remote field in client export if the system's FQDN changes as @gsanchietti  wrote.

Said that I think @steph modification is a good improvement over actual behaviour, to increase admin awareness I would eventually suggest to remove  automatic writing of FQDN on the db prop if the field is empty and I would add a validator to require the field is not empty.
In this way, the admin is forced to write the value he wants to use.

IMHO The positives of this change far outweigh the negatives: strange behaviour in the GUI is fixed, the FQDN change problem remain, but I consider it to be much less critical,both because FQDN change is very rare (much than the changes to the openvpn server) and also because we already have this problem in actual implementation, if we saved a value (an IP address or a FQDN) in the box.



To summarize I propose to take @stephdl proposal with these modifications:
->remove set prop automatic if empty (leave placeholder)
-> force admin to write something in the box


It can easily work with servers already in production:
In case the prop is empty use FQDN, at first modification you will be forced to write the value, if you don’t know the value, the placeholder will suggest you the right thing.

**Comment by gsanchietti** [comment]
This could be a good compromise!

**Comment by stephdl** [comment]
The public engine is still there to retrieve the external IP, if you erase the proposed IP, the placeholder is displayed with the FQDN, you cannot save an empty textarea, you have to write either an IP or a hostname

**Comment by stephdl** [comment]
However we have a new issue : `    $v->declareParameter('Remote', Validate::ANYTHING);`

**Review by gsanchietti — CHANGES_REQUESTED** [review]
If the validator gives an error, the UI should report the error description in red color just below the textarea.
Please keep the behavior consistent with all other fields.

**Comment by stephdl** [comment]
it should be good @gsanchietti, the validator is done at the api level

**Review by gsanchietti — APPROVED** [review]
Good. ;)
Open the related issue before merging.
